velox/connectors/hive/HiveDataSink.h

```diff
@@ -79,6 +79,104 @@ class LocationHandle : public ISerializable {
   const TableType tableType_;
 };
 
+class HiveSortingColumn : public ISerializable {
+ public:
+  HiveSortingColumn(
+      const std::string& sortColumn,
+      const core::SortOrder& sortOrder);
+
+  const std::string& sortColumn() const {
+    return sortColumn_;
+  }
+
+  core::SortOrder sortOrder() const {
+    return sortOrder_;
+  }
+
+  folly::dynamic serialize() const override;
+
+  static std::shared_ptr<HiveSortingColumn> deserialize(
+      const folly::dynamic& obj,
+      void* context);
+
+  std::string toString() const;
+
+  static void registerSerDe();
+
+ private:
+  const std::string sortColumn_;
+  const core::SortOrder sortOrder_;
+};
+
+class HiveBucketProperty : public ISerializable {
+ public:
+  enum class Kind { HIVE_COMPATIBLE, PRESTO_NATIVE };
+
+  HiveBucketProperty(
+      Kind kind,
+      int32_t bucketCount,
+      const std::vector<std::string>& bucketedBy,
+      const std::vector<TypePtr>& bucketedTypes,
+      const std::vector<std::shared_ptr<const HiveSortingColumn>>& sortedBy);
```

Comment:
Why not just std::vector<HiveSortingColumn> ? Do we need to wrap HiveSortingColumn in a shared pointer?

Reply:
It is ease dynamic object deserialization handling. Currently we only support shared_ptr/unique_ptr object creation through deserialization.

Reply:
Is this really so? I don't think such limitation exists.

Reply:
See WindowNode for an example.